inspirehep/modules/authors/rest/stats.py

```diff
@@ -0,0 +1,124 @@
+#
+# In applying this licence, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+from __future__ import absolute_import, division, print_function
+
+from collections import Counter
+import json
+
+from inspirehep.modules.search import LiteratureSearch
+from inspirehep.utils.stats import calculate_h_index, calculate_i10_index
+
+
+class AuthorAPIStats(object):
+    """API endpoint for author collection returning statistics."""
+
+    def serialize(self, pid, record, links_factory=None):
+        """Return a different metrics for a given author recid.
+
+        :param pid:
+            Persistent identifier instance.
+
+        :param record:
+            Record instance.
+
+        :param links_factory:
+            Factory function for the link generation, which are added to
+            the response.
+        """
+        author_pid = pid.pid_value
+
+        fields = set()
+        keywords = []
+
+        statistics = {}
+        statistics['citations'] = 0
+        statistics['publications'] = 0
+        statistics['types'] = {}
+
+        statistics_citations = {}
+
+        search = LiteratureSearch().query({
+            "match": {
+                "authors.recid": author_pid
+            }
+        }).params(
+            _source=[
+                "citation_count",
+                "control_number",
+                "facet_inspire_doc_type",
+                "facet_inspire_subjects",
+                "thesaurus_terms",
+            ]
+        )
+
+        for result in search.scan():
+            result_source = result.to_dict()
+
+            # Increment the count of the total number of publications.
+            statistics['publications'] += 1
+
+            # Increment the count of citations.
+            citation_count = result_source.get('citation_count', 0)
+
+            statistics['citations'] += citation_count
+            statistics_citations[int(result_source['control_number'])] = \
+                citation_count
+
+            # Count how many times certain type of publication was published.
+            try:
+                publication_type = result_source.get(
+                    'facet_inspire_doc_type', [])[0]
+            except IndexError:
+                pass
+
+            if publication_type:
+                if publication_type in statistics['types']:
+                    statistics['types'][publication_type] += 1
+                else:
+                    statistics['types'][publication_type] = 1
+
+            # Get fields.
+            for field in result_source.get('facet_inspire_subjects', []):
+                fields.add(field)
+
+            # Get keywords.
+            for term in result_source.get('thesaurus_terms', []):
+                if term.get('keyword') is not "* Automatic Keywords *":
+                    keywords.append(term.get('keyword'))
```

Comment:
Also note this thing about Python:
```python
In [1]: s = 'foo'

In [2]: t = ''.join(['f', 'o', 'o'])

In [3]: s == t
Out[3]: True

In [4]: s is t
Out[4]: False
```
comparing strings with `is` is risky in very subtle ways.